ext/date/php_date.c

```diff
@@ -2380,6 +2383,34 @@ static zend_object *date_object_clone_timezone(zval *this_ptr) /* {{{ */
 	return &new_obj->std;
 } /* }}} */
 
+static int date_object_compare_timezone(zval *tz1, zval *tz2) /* {{{ */
+{
+	php_timezone_obj *o1, *o2;
+
+	o1 = Z_PHPTIMEZONE_P(tz1);
+	o2 = Z_PHPTIMEZONE_P(tz2);
+
+	if (!o1->initialized || !o2->initialized) {
+		php_error_docref(NULL, E_WARNING, "Trying to compare an incomplete DateTimeZone object");
```

Comment:
If this happens, there is some weird stuff going on :-) I bet you can't make a test case for it.

Reply:
I've just put that in because I was not sure if it can happen. Probably best to change that to an assertion, if it's not supposed to happen.